Modules/Wishlist/Http/Controllers/WishlistController.php

```diff
@@ -20,63 +22,201 @@ public function __construct(WishlistService $wishlistService)
     }
 
     /**
+     * Add wishlist by logged user
+     *
      * @param Request $request
      * @return JsonResponse
      */
     public function add(Request $request): JsonResponse
     {
+        $user = auth()->user();
+        if ($user->cannot('create', Wishlist::class)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
         $response = $this->wishlistService->add($request);
+
         return response()->json($response);
     }
 
     /**
+     * Fetch wishlist by logged user
+     *
      * @param Request $request
      * @return JsonResponse
      */
     public function fetch(Request $request): JsonResponse
```

Comment:
$request is unnecessary parameter, remove.

Reply:
Done